src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,154 @@
+                if (descriptionExists(inputArr, "Oops, an event description cannot be left empty!")) {
+                    String[] descrArr = description.split(" /at ");
+                    if (keywordExists(descrArr, "/at", "event")) {
+                        LocalDateTime dateTime = parseDateTime(descrArr[1]);
+                        Task task = new Event(descrArr[0], dateTime);
+                        return new AddCommand(task);
+                    }
+                }
+                break;
+            case "find":
+                if (descriptionExists(inputArr, "Please specify a keyword!")) {
+                    if (inputArr.length > 2) {
+                        throw new DukeException("Please specify only one keyword!");
+                    } else {
+                        return new FindCommand(description);
+                    }
+                }
+                break;
+            case "bye":
+                return new ExitCommand();
+            default:
+                throw new DukeException("Oh no! I don't understand what that means...");
+            }
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid date/time format.\n    "
+                    + "Please use the following format: yyyy-mm-dd HH:mm");
+        }
+        return new InvalidCommand();
+    }
+
+    /**
+     * Returns a boolean indicating whether there is a task description provided in the user input.
+     *
+     * @param inputArr The user input split by spaces.
+     * @param errorMessage The error message to be printed if description does not exist.
+     * @return True if there is a description, false otherwise.
+     * @throws DukeException If the description is not provided.
+     */
+    public static boolean descriptionExists(String[] inputArr, String errorMessage) throws DukeException {
+        if (inputArr.length == 1) {
+            throw new DukeException(errorMessage);
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Returns a boolean indicating whether there is a given keyword provided in the task description.
+     *
+     * @param descrArr The description split by the keyword.
+     * @param keyword The keyword to be found in the description.
+     * @param taskType The type of the task (e.g. deadline, event)
+     * @return True if keyword can be found, false otherwise.
+     * @throws DukeException If keyword is not found in description.
+     */
+    public static boolean keywordExists(String[] descrArr, String keyword, String taskType) throws DukeException {
+        if (descrArr.length == 1) {
+            throw new DukeException("Oops, please use " + keyword
+                    + " to set a date and time for this " + taskType + "!");
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Returns LocalDateTime.
+     *
+     * @param dt The string of format yyyy-MM-dd HH:mm
+     * @return LocalDateTime.
```

Comment:
Perhaps could elaborate on what `LocalDateTime` it is? i.e. the parsed `LocalDateTime`?